src/goto-programs/interpreter.cpp

```diff
@@ -500,7 +500,8 @@ exprt interpretert::get_value(
     }
     return result;
   }
-  if(use_non_det && (memory[offset].initialized>=0))
+  if(use_non_det &&
+     memory[offset].initialized!=memory_cellt::initializedt::WRITTEN_BEFORE_READ)
```

Comment:
@dcattaruzza Are you saying that this new version is correct? Note that it vastly differs from the previous one as sketched by @mgudemann.